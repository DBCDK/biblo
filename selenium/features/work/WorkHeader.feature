Feature: WorkHeader

  Scenario: Title should be a link pointing to the given material
    When mock IMessisFodspor is loaded
    Given a user visits the reviews page with review_id 79285
    When the cookiewarning has been closed
    And I click on the element .work-detail--title-container
    Then the path should be /materiale/870970-basis:50911713

  Scenario: Coverimage should be wrapped in a link pointing to the given material
    When mock IMessisFodspor is loaded
    Given a user visits the reviews page with review_id 79285
    When the cookiewarning has been closed
    And I click on the element .work-header--foreground-image
    Then the path should be /materiale/870970-basis:50911713
