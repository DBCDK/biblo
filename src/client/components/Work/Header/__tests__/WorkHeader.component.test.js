/**
 * @file
 * Unittesting methods in WorkHeader.component.test.js
 */

import React from 'react';
import {shallow} from 'enzyme';

import {WorkHeader} from './../WorkHeader.component';

describe('Testing the WorkHeader.component', () => {
  it('Matches the snapshot', () => {
    const wrapper = shallow(<WorkHeader coverUrl={''} linkToMaterial={''} />);
    expect(wrapper).toMatchSnapshot();
  });
});
