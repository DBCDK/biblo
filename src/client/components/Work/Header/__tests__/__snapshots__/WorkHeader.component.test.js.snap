// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the WorkHeader.component It renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorkHeader
    coverUrl=""
    linkToMaterial=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="work-header--background-image "
          style={
            Object {
              "backgroundImage": "url(\\"\\")",
            }
          }
        />,
        <div
          className="work-header--foreground-image--wrapper "
        >
          <div
            className="work-header--foreground-image "
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <a
              href=""
            >
              <img
                className=""
                onLoad={[Function]}
                src=""
              />
            </a>
          </div>
        </div>,
      ],
      "className": "work-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "work-header--background-image ",
          "style": Object {
            "backgroundImage": "url(\\"\\")",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="work-header--foreground-image "
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <a
              href=""
            >
              <img
                className=""
                onLoad={[Function]}
                src=""
              />
            </a>
          </div>,
          "className": "work-header--foreground-image--wrapper ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href=""
            >
              <img
                className=""
                onLoad={[Function]}
                src=""
              />
            </a>,
            "className": "work-header--foreground-image ",
            "style": Object {
              "marginTop": "40px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className=""
                onLoad={[Function]}
                src=""
              />,
              "href": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "onLoad": [Function],
                "src": "",
              },
              "ref": [Function],
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="work-header--background-image "
            style={
              Object {
                "backgroundImage": "url(\\"\\")",
              }
            }
          />,
          <div
            className="work-header--foreground-image--wrapper "
          >
            <div
              className="work-header--foreground-image "
              style={
                Object {
                  "marginTop": "40px",
                }
              }
            >
              <a
                href=""
              >
                <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />
              </a>
            </div>
          </div>,
        ],
        "className": "work-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "work-header--background-image ",
            "style": Object {
              "backgroundImage": "url(\\"\\")",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="work-header--foreground-image "
              style={
                Object {
                  "marginTop": "40px",
                }
              }
            >
              <a
                href=""
              >
                <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />
              </a>
            </div>,
            "className": "work-header--foreground-image--wrapper ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href=""
              >
                <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />
              </a>,
              "className": "work-header--foreground-image ",
              "style": Object {
                "marginTop": "40px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />,
                "href": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "",
                  "onLoad": [Function],
                  "src": "",
                },
                "ref": [Function],
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Testing the WorkHeader.component Matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorkHeader
    coverUrl=""
    linkToMaterial=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="work-header--background-image "
          style={
            Object {
              "backgroundImage": "url(\\"\\")",
            }
          }
        />,
        <div
          className="work-header--foreground-image--wrapper "
        >
          <div
            className="work-header--foreground-image "
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <a
              href=""
            >
              <img
                className=""
                onLoad={[Function]}
                src=""
              />
            </a>
          </div>
        </div>,
      ],
      "className": "work-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "work-header--background-image ",
          "style": Object {
            "backgroundImage": "url(\\"\\")",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="work-header--foreground-image "
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <a
              href=""
            >
              <img
                className=""
                onLoad={[Function]}
                src=""
              />
            </a>
          </div>,
          "className": "work-header--foreground-image--wrapper ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href=""
            >
              <img
                className=""
                onLoad={[Function]}
                src=""
              />
            </a>,
            "className": "work-header--foreground-image ",
            "style": Object {
              "marginTop": "40px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className=""
                onLoad={[Function]}
                src=""
              />,
              "href": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "onLoad": [Function],
                "src": "",
              },
              "ref": [Function],
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="work-header--background-image "
            style={
              Object {
                "backgroundImage": "url(\\"\\")",
              }
            }
          />,
          <div
            className="work-header--foreground-image--wrapper "
          >
            <div
              className="work-header--foreground-image "
              style={
                Object {
                  "marginTop": "40px",
                }
              }
            >
              <a
                href=""
              >
                <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />
              </a>
            </div>
          </div>,
        ],
        "className": "work-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "work-header--background-image ",
            "style": Object {
              "backgroundImage": "url(\\"\\")",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="work-header--foreground-image "
              style={
                Object {
                  "marginTop": "40px",
                }
              }
            >
              <a
                href=""
              >
                <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />
              </a>
            </div>,
            "className": "work-header--foreground-image--wrapper ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href=""
              >
                <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />
              </a>,
              "className": "work-header--foreground-image ",
              "style": Object {
                "marginTop": "40px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className=""
                  onLoad={[Function]}
                  src=""
                />,
                "href": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "",
                  "onLoad": [Function],
                  "src": "",
                },
                "ref": [Function],
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
